### PR TITLE
VIH-9352 add edit justice user feature

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###
VIH-9352

### Change description ###
* use existing 'add user' dialog to allow user to edit a user's role
* disable all form fields apart from Role
* add new EditJusticeUser endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                x.EditJusticeUserAsync(It.Is<EditJusticeUserRequest>(r => r.Id == expectedResponse.Id)), Times.Once());
+            result.Should().BeOfType<OkObjectResult>().And.Subject.As<OkObjectResult>().Value.Should()
+                .Be(expectedResponse);
+        }
+
+        [Test]
+        public async Task should_forward_bad_request_from_bookings_api_to_client_app()
+        {
+            // arrange
+            var validationProblemDetails = new ValidationProblemDetails(new Dictionary<string, string[]>
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs:77]: Nice!

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/justice-user-form/justice-user-form.component.ts [hunk 2]
```diff
@@ -47,21 +51,37 @@ export class JusticeUserFormComponent {
...
-                this.form.controls.firstName.value,
-                this.form.controls.lastName.value,
-                this.form.controls.contactTelephone.value,
-                this.form.value.role
-            )
-            .subscribe({
-                next: newJusticeUser => this.onSaveSucceeded(newJusticeUser),
-                error: (error: string | BookHearingException) => this.onSaveFailed(error)
-            });
+        if (this.mode === 'add') {
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/justice-user-form/justice-user-form.component.ts:64]: Sorry to be pedantic, but would you mind breaking the blocks into their own methods so the onSave can be leaner and more verbose

```
onSave() {
 // code excluded
 if (this.mode === 'add') {
   this.addNewUser()
 } 
 else if (this.mode === 'edit') {
      this.updateExistingUser()
 }
}
```

**Review by shaed-parkar — APPROVED** [review]
Looks good though I have one suggestion

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/justice-user-form/justice-user-form.component.ts (hunk 2 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/justice-user-form/justice-user-form.component.ts:64]: Done, thanks

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        {
+            _mocker = AutoMock.GetLoose();
+            _sut = SetupControllerWithClaims();
+        }       
+
+        private JusticeUsersController SetupControllerWithClaims()
+        {
+            var cp = new ClaimsPrincipalBuilder().WithRole(AppRoles.VhOfficerRole)
+                .WithUsername(_username).Build();
+            var context = new ControllerContext
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs:38]: I don't think we need to set the claims here, as the user identity isn't being referenced in the controller action.

We should be able to simplify this to:

```
private JusticeUsersControlloer SetupController()
{
  var controller = _mocker.Create<JusticeUsersController>();
  return controller;
}
```

**Review by shaed-parkar — APPROVED** [review]

**Review by oliver-scott — COMMENTED** [review]
really good, just one suggestion

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs (hunk 3 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs]: Ah, nice - thanks Oliver. Done.

Comment (oliver-scott) [reply AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs]: great. A couple of other things to follow on from that:

- Maybe rename the method to `SetupController`, as it no longer sets up claims.
- `private readonly string _username = "test.vho@hmcts.net";` (line 25 of the same file) - this is no longer used, so we should be able to remove that.

Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite.UnitTests/Controllers/JusticeUserController/EditJusticeUserTests.cs]: Got it - done. Thanks!

**Review by will-craig — APPROVED** [review]

**Review by oliver-scott — APPROVED** [review]

**Pull request merged**